src/util/FormUtils.js

```diff
@@ -478,4 +478,103 @@ export default class FormUtils {
       recursiveTraverse(q);
     });
   }
+
+  /**
+   * Performs a dfs on a set of questions and executes callback functions
+   * when entering and leaving each question.
+   * @param {Array<Object>} questions - The array of questions to traverse.
+   * @param {function} onEnterQuestion - The callback function to execute when entering a question.
+   * @param {function} onLeaveQuestion - The callback function to execute when leaving a question.
+   */
+  static dfsTraverseQuestionTreeExtended(
```

Comment:
I do not see the reason to define new method. Citing myself from the issue description:
"""
Thus I suggest adding `FormUtils.printFormSpecification` that would use modified dfsTraverse function. The modified dfsTraverse function would have 3 parameters, while 3. parameter should be optional, i.e.
`dfsTraverse(startingQuestions:array, onEnterQuestion:function, onLeaveQuestion:function)`
"""

Reply:
It makes sense to me to just modify the existing one while usage does not need to be refactored as the modification is backward compatible as 3rd param is optional (i.e. defaulting to the empty method -- (question) => {}) .

Reply:
@LaChope

Reply:
Agree